lib/frrscript.h

```diff
@@ -149,53 +177,81 @@ const struct prefix * : lua_decode_noop                         \
 )(L, -1, value)
 
 /*
- * Call script.
+ * Call Lua function state (abstraction for a single Lua function)
  *
- * fs
- *    The script to call; this is obtained from frrscript_load().
+ * lfs
+ *    The Lua function to call; this should have been loaded in by
+ * frrscript_load(). nargs Number of arguments the function accepts
  *
  * Returns:
  *    0 if the script ran successfully, nonzero otherwise.
  */
-int _frrscript_call(struct frrscript *fs);
+int _frrscript_call_lua(struct lua_function_state *lfs, int nargs);
 
 /*
- * Wrapper for call script. Maps values passed in to their encoder
- * and decoder types.
+ * Wrapper for calling Lua function state. Maps values passed in to their
+ * encoder and decoder types.
  *
  * fs
- *    The script to call; this is obtained from frrscript_load().
+ *    The struct frrscript in which the Lua fuction was loaded into
+ * f
+ *    Name of the Lua function.
  *
  * Returns:
  *    0 if the script ran successfully, nonzero otherwise.
  */
-#define frrscript_call(fs, ...)                                                \
-	({                                                                     \
-		lua_State *L = fs->L;                                          \
-		MAP_LISTS(ENCODE_ARGS, ##__VA_ARGS__);                         \
-		int ret = _frrscript_call(fs);                                 \
-		if (ret == 0) {                                                \
-			MAP_LISTS(DECODE_ARGS, ##__VA_ARGS__);                 \
-		}                                                              \
-		ret;                                                           \
+#define frrscript_call(fs, f, ...)                                                                                                                                 \
+	({                                                                                                                                                         \
+		struct lua_function_state lookup = {.name = f};                                                                                                    \
+		struct lua_function_state *lfs;                                                                                                                    \
+		lfs = hash_lookup(fs->lua_function_hash, &lookup);                                                                                                 \
```

Comment:
When you use macro arguments you need to wrap them in (parens). This is because expressions can be passed as arguments and when expanded in a macro can have unintended behavior due to order of operations and other effects.

Reply:
updated in https://github.com/FRRouting/frr/pull/8982/commits/67b64027b2a5fc97d58f1298e5b99da1bbe78a68